### PR TITLE
SMAP regression error

### PR DESCRIPTION
**Pull request by hugoperrin**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
This PR adds SMAPE regression metric: https://github.com/PyTorchLightning/metrics/issues/372.

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/helpers/non_sklearn_metrics.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+"""File for non sklearn metrics that are to be used for reference for tests"""
+
+import numpy as np
+from sklearn.metrics._regression import _check_reg_targets
+from sklearn.utils.validation import check_consistent_length
+
+
+def symmetric_mean_absolute_percentage_error(y_true, y_pred, sample_weight=None, multioutput='uniform_average'):
+    """Symmetric mean absolute percentage error regression loss.
```
Comment (SkafteNicki) [inline tests/helpers/non_sklearn_metrics.py:9]: could you insert a reference here just to be sure which definition you used, if we need to debug in the future

File: tests/helpers/non_sklearn_metrics.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+"""File for non sklearn metrics that are to be used for reference for tests"""
+
+import numpy as np
+from sklearn.metrics._regression import _check_reg_targets
+from sklearn.utils.validation import check_consistent_length
+
+
+def symmetric_mean_absolute_percentage_error(y_true, y_pred, sample_weight=None, multioutput='uniform_average'):
```
Comment (SkafteNicki) [inline tests/helpers/non_sklearn_metrics.py:8]:
```suggestion
def symmetric_mean_absolute_percentage_error(y_true: np.ndarray, y_pred: np.ndarray, sample_weight: Optional[np.ndarray]=None, multioutput: str='uniform_average'):
```

File: tests/helpers/non_sklearn_metrics.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+"""File for non sklearn metrics that are to be used for reference for tests"""
+
```
Comment (SkafteNicki) [inline tests/helpers/non_sklearn_metrics.py:2]: for the function typing above
```suggestion
from typing import Optional
```

File: torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    return sum_abs_per_error, num_obs
+
+
+def _symmetric_mean_absolute_percentage_error_compute(sum_abs_per_error: Tensor, num_obs: int) -> Tensor:
+    return sum_abs_per_error / num_obs
+
+
+def symmetric_mean_absolute_percentage_error(preds: Tensor, target: Tensor) -> Tensor:
+    """
+    Computes symmetric mean absolute percentage error.
```
Comment (SkafteNicki) [inline torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py:46]: insert link to wiki page (https://en.wikipedia.org/wiki/Symmetric_mean_absolute_percentage_error) + formula

File: torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    Args:
+        preds: estimated labels
+        target: ground truth labels
+
+    Return:
+        Tensor with SMAPE
+
+    Note:
```
Comment (SkafteNicki) [inline torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py:55]: this note does not really make sense to the user without knowing how the epsilon influences the computation. I suggest removing it

**Review by SkafteNicki — COMMENTED** [review]
Looks great. Two things:
* Remember to add an note in the `Changelog` file
* The function that is compared against takes a `multioutput` argument but the functional/modular implementation of SMAPE does not. Why is that?

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@hugoperrin mind check last comments from @SkafteNicki, mainly related to references

File: tests/helpers/non_sklearn_metrics.py (hunk 1 above)
Comment (hugoperrin) [reply tests/helpers/non_sklearn_metrics.py:15]: I've added the wiki page and the formula for reference

File: torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py (hunk 4 above)
Comment (hugoperrin) [reply torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py:46]: I added them :)

File: torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py (hunk 5 above)
Comment (hugoperrin) [reply torchmetrics/functional/regression/symmetric_mean_absolute_percentage_error.py]: I removed it indeed. I originally kept it, in order to explain the default value used for epsilon, but it's true it does not make much sense since I'm not comparing it to a sklearn implementation

**Comment by hugoperrin** [comment]
@Borda I updated the needed docs, I think this should be good, I'll wait on the CI to finish and fix things if things are needed to be fixed then :) Thank you for your quick feedback :)

**Comment by Borda** [comment]
> PS: should I squash the commits by the way?

no need, well do it with all PR by default: squash + merge

**Review by SkafteNicki — APPROVED** [review]
LGTM
